src/main/kotlin/com/kreait/docs/common/AsciiDocUtils.kt

```diff
@@ -0,0 +1,81 @@
+package com.kreait.docs.common
+
+import com.kreait.docs.data.Guide
+import org.asciidoctor.Asciidoctor
+import org.asciidoctor.Options
+import org.asciidoctor.OptionsBuilder
+import org.asciidoctor.SafeMode
+import org.asciidoctor.jruby.AsciiDocDirectoryWalker
+import org.springframework.stereotype.Component
+import java.io.File
+import java.lang.IllegalArgumentException
+
+@Component
+class AsciiDocUtils {
+
+    private var asciiDoctor = Asciidoctor.Factory.create()
+    private val guideList = mutableListOf<Guide>()
+
+    fun walkAndSort(dir: String, zipFile: File?, unzippedRoot: File?): MutableList<Guide> {
+
+        val walker = AsciiDocDirectoryWalker(dir)
+        walker.forEach { file ->
+            val options = OptionsBuilder.options().baseDir(unzippedRoot)
+            guideList.add(Guide(
+                    id = file.nameWithoutExtension,
+                    title = "${asciiDoctor.loadFile(file, options.asMap()).getAttribute("title")}",
```

Comment:
asciiDoctor.loadFile(file, options.asMap()) can be saved in a variable, so you dont have to load it always